trio_websocket/_impl.py

```diff
@@ -137,14 +168,90 @@ async def open_websocket(
             raise ConnectionTimeout from None
         except OSError as e:
             raise HandshakeError from e
+
+    async def close_connection(connection: WebSocketConnection) -> None:
         try:
-            yield connection
-        finally:
-            try:
-                with trio.fail_after(disconnect_timeout):
-                    await connection.aclose()
-            except trio.TooSlowError:
-                raise DisconnectionTimeout from None
+            with trio.fail_after(disconnect_timeout):
+                await connection.aclose()
+        except trio.TooSlowError:
+            raise DisconnectionTimeout from None
+
+    if _TRIO_MULTI_ERROR:
+        exception_group_type = trio.MultiError  # type: ignore[attr-defined] # pylint: disable=no-member
+    else:
+        exception_group_type = BaseExceptionGroup
+
+    connection: WebSocketConnection|None=None
+    result2: outcome.Maybe[None] | None = None
```

Comment:
I forgot if we discussed already, but I think `Optional[Foo]` is more friendly / readable than `Foo | None`.

Reply:
I think `|None` is the standard way of doing it nowadays, and pyupgrade/ruff will warn on doing `Optional[...]`: https://docs.astral.sh/ruff/rules/non-pep604-annotation/

I also think it's nice for newbies not to have another keyword to know, and highly personally I prefer not having to do another import for it.